main/templates/main/index.html

```diff
@@ -1,7 +1,7 @@
 {% extends 'main/base.html'%}
 
 {% block content %}
-    {% for post in posts %}
+    {% for post in object_list %}
```

Comment:
It is possible to rename the queryset result objects name for template context in class-based views. Do this.